### ISSUE TITLE
Pattern `CandyCloud` is crashy

### ISSUE THREAD

**Issue by meawoppl** [squaredproject/Entwined#25]
Often when playing with parameters, I get this traceback:
```
[LX 2020/11/29 13:30:28] Unexpected exception in device loop com.charlesgadeken.entwined.patterns.contributors.kyleFleming.CandyCloudTextureEffect: Index: 690, Size: 690
java.lang.IndexOutOfBoundsException: Index: 690, Size: 690
        at java.util.ArrayList.rangeCheck(ArrayList.java:659)
        at java.util.ArrayList.get(ArrayList.java:435)
        at java.util.Collections$UnmodifiableList.get(Collections.java:1311)
        at com.charlesgadeken.entwined.patterns.contributors.kyleFleming.CandyCloudTextureEffect.run(CandyCloudTextureEffect.java:26)
        at heronarts.lx.effect.LXEffect.onLoop(LXEffect.java:286)
        at heronarts.lx.LXLayeredComponent.loop(LXLayeredComponent.java:119)
        at heronarts.lx.LXDeviceComponent.loop(LXDeviceComponent.java:146)
        at com.charlesgadeken.entwined.effects.EntwinedBaseEffect.loop(EntwinedBaseEffect.java:21)
        at heronarts.lx.mixer.LXMixerEngine.loop(LXMixerEngine.java:985)
        at heronarts.lx.LXEngine._run(LXEngine.java:854)
        at heronarts.lx.LXEngine.run(LXEngine.java:750)
        at heronarts.lx.LXEngine.access$900(LXEngine.java:59)
        at heronarts.lx.LXEngine$EngineThread$RenderTask.run(LXEngine.java:622)
        at java.util.TimerThread.mainLoop(Timer.java:555)
        at java.util.TimerThread.run(Timer.java:505)
```

**Comment by meawoppl** [comment]
@kylefleming can you fix this in the newlx tree also?

**Comment by bbulkow** [comment]
Some of the patterns crashed because there was a calculation that depended on the number of cubes, and if you're not getting that right, boom. Wonder if that's going on here.

**Comment by meawoppl** [comment]
It seems linked to one of it parameters, but I haven't spent any cycles in
triage...

**Comment by meawoppl** [comment]
It is specifically tied to the track knob wired onto `CandyCloudTextureEffect`

**Comment by meawoppl** [comment]
Appears to have been fixed by #40
